### PR TITLE
Update haproxy-best-practices.md

### PR DESCRIPTION
**Pull request by Chad20N13**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
maxconn Parameter

maxconn Parameter description is incorrect.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?
https://github.com/pingcap/docs/pull/7046

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/7805
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -169,7 +169,7 @@ global                                     # Global configuration.
    log         127.0.0.1 local2            # Global syslog servers (up to two).
    chroot      /var/lib/haproxy            # Changes the current directory and sets superuser privileges for the startup process to improve security.
    pidfile     /var/run/haproxy.pid        # Writes the PIDs of HAProxy processes into this file.
-   maxconn     256                         # The maximum number of concurrent connections per HAProxy thread.
+   maxconn     4096                        # The maximum per-process number of concurrent connections to <4096>. It is equivalent to the command-line argument "-n".
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:172]: Shall we keep the English version as close to the Chinese version `单个 HAProxy 进程可接受的最大并发连接数。` as possible? If `It is equivalent to the command-line argument "-n".` is also important, please add it to the [Chinese version](https://github.com/pingcap/docs-cn/pull/7805/files#diff-241fc1deb2dc5d2b5b537b770bed50afff6018b1ecd80927f7e125a8ec6072b7R171). Thanks!

Comment (TomShawn) [reply best-practices/haproxy-best-practices.md:172]:
```suggestion
maxconn 4096 # The maximum number of concurrent connections for a single HAProxy process. It is equivalent to the command-line argument "-n".
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Review by superlzs0476 — APPROVED** [review]
LGTM

**Pull request merged**